DataFormats/TrackerRecHit2D/interface/SiTrackerMultiRecHit.h

```diff
@@ -34,7 +34,7 @@ class SiTrackerMultiRecHit : public BaseTrackerRecHit {
   // at the momement nobody care of MultiHit!!!
   // used by trackMerger (to be improved)
   OmniClusterRef const& firstClusterRef() const override {
-    return static_cast<BaseTrackerRecHit const*>(&theHits.front())->firstClusterRef();
+    return dynamic_cast<BaseTrackerRecHit const&>(theHits.front()).firstClusterRef();
```

Comment:
this will abort! either test the result or assume is correct by costruction

Reply:
I'm reverting this to a static_cast so the subject is moot, but philosophically I disagree.  If a check finds a logic error in code that is supposed to be correct by construction, throwing an exception is a reasonable response.

Reply:
Well, CMS could finally decide to have two builds: validation )(with debug on) and production (no debug).
As long as  CMS does not switch to such a sound (to me) practice, we must enforce maximal efficiency.
I am also pretty annoyed that upgrade is polluting production code  that is supposed to run in production for Run3. Most probably NONE of this code will ever made it (I hope!) to Phase2,
still upgrade is finding her way in the most old and fragile components of cms software w/o any prior verification of the consequences upstream and downstream.
I sincerely do not know WHO has approved all this MTD code in Tracker and I have missed any design discussion of all this.

Reply:
MTD had to write a TDR. Subsequently, they have not devoted much time to code maintenance or improvement tasks (currently ~40% of the upgrade issues on GitHub are related to MTD).

If there are issues that aren't being caught by our current tests, more tests should be added.